### PR TITLE
AZ625 - Simplify 2I HTTP interface

### PR DESCRIPTION
**Pull request by rustyio**
TL;DR - For simplicity, this code removes some 2i operations from HTTP interface, bringing it into parity with the MapReduce and PB interfaces.
# Setup

**Enable the index backend before running!**
1. `cd rel/riak`
2. `$EDITOR etc/app.config`
3. `Change storage_backend to riak_kv_index_backend`
# Overview

The first round of changes to the HTTP interface for 2I added support for:
- equality
- range
- greater than
- less than
- greater than or equal
- less than or equal operations.

For simplicity, we only support equality and range operations in the Map/Reduce and PB interfaces.

This change **removes** everything except for equality and range from the HTTP interface. It slightly reduces code complexity, but will make documentation easier if all interface support the same operations.

In the future, we will likely add back more complex operations through a declarative query language.
# Testing

``` bash
# Index some documents...

curl -v -X PUT \
-d 'data1' \
-H "Content-Type: application/json" \
-H "x-riak-index-field1_bin: val1" \
-H "x-riak-index-field2_int: 1001" \
http://127.0.0.1:8098/riak/mybucket/mykey1

curl -v -X PUT \
-d 'data2' \
-H "Content-Type: application/json" \
-H "x-riak-index-Field1_bin: val2" \
-H "x-riak-index-Field2_int: 1002" \
http://127.0.0.1:8098/riak/mybucket/mykey2

curl -v -X PUT \
-d 'data3' \
-H "Content-Type: application/json" \
-H "X-RIAK-INDEX-FIELD1_BIN: val3" \
-H "X-RIAK-INDEX-FIELD2_INT: 1003" \
http://127.0.0.1:8098/riak/mybucket/mykey3

curl -v -X PUT \
-d 'data4' \
-H "Content-Type: application/json" \
-H "x-riak-index-field1_bin: val4" \
-H "x-riak-index-field2_int: 1004" \
http://127.0.0.1:8098/riak/mybucket/mykey4

# Retrieve the documents...

curl -i http://127.0.0.1:8098/riak/mybucket/mykey1
curl -i http://127.0.0.1:8098/riak/mybucket/mykey2
curl -i http://127.0.0.1:8098/riak/mybucket/mykey3
curl -i http://127.0.0.1:8098/riak/mybucket/mykey4

# Query against a binary field...

curl http://localhost:8098/buckets/mybucket/index/field1_bin/val1
curl http://localhost:8098/buckets/mybucket/index/field1_bin/val2/val4

# Query against an integer field...

curl http://localhost:8098/buckets/mybucket/index/field2_int/1001
curl http://localhost:8098/buckets/mybucket/index/field2_int/1002/1004

# Should throw errors: too few / too many args...

curl http://localhost:8098/buckets/mybucket/index/field2_int
curl http://localhost:8098/buckets/mybucket/index/field2_int/1/2/3
```

### CODE REVIEW

**Comment by slfritchie** [comment]
Rusty: I see the following with the first binary field query,
`curl http://localhost:8098/buckets/mybucket/index/field1_bin/val1`

```
16:03:10.446 [error] gen_fsm <0.397.0> in state active terminated with reason: call to undefined function riak_kv_bitcask_backend:fold_index/3 from riak_kv_vnode:index_query/6
16:03:10.448 [error] CRASH REPORT Process [] with 0 neighbours crashed with reason: call to undefined function riak_kv_bitcask_backend:fold_index({#Ref<0.0.0.1273>,[46,47,100,97,116,97,47,98,105,116,99,97,115,107,47,54,51,57,52,48,54,57,54,54,51,51,50,50,55,48,48,50,54,55,49,52,49,49,50,49,49,52,51,49,51,51,55,51,56,50,49,48,57,57,52,55,48,52,56,55,53,53,50]}, <<109,121,98,117,99,107,101,116>>, {eq,<<102,105,101,108,100,49,95,98,105,110>>,[<<118,97,108,49>>]})
16:03:10.449 [error] gen_fsm <0.391.0> in state active terminated with reason: call to undefined function riak_kv_bitcask_backend:fold_index/3 from riak_kv_vnode:index_query/6
```

... for a total of 66 error messages.  Curl gets a 500 HTTP response a minute later.

**Comment by slfritchie** [comment]
Once the mundane/perhaps-pilot-error problems are ironed out for code review,
this is a thing that is dying to be basho_expect'ified.  Please try it out, using the
following as guideposts:
- basho_expect/systest_prepost_hooks.py
- basho_expect/wrap_systest_prepost_hooks.py
- Run-systest-prepost-hooks

**Comment by slfritchie** [comment]
Thanks for the setup instructions.  +1 to merge, though I think that a basho_expect
test for this feature is mandatory.  (Other 2I stuff should also be tested by b_e.)
Please create a new kanban card for the task, put a deadline on it of one week, and
then hang it in the Ready column (instead of the usual Backlog).
